src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -158,10 +158,28 @@ public DecompilationMetadataAsSourceFileProvider(IImplementationAssemblyLookupSe
                 // Create the directory. It's possible a parallel deletion is happening in another process, so we may have
                 // to retry this a few times.
                 var directoryToCreate = Path.GetDirectoryName(fileInfo.TemporaryFilePath)!;
+                var stopwatch = SharedStopwatch.StartNew();
+                var firstAttempt = true;
                 while (!Directory.Exists(directoryToCreate))
                 {
+                    if (stopwatch.Elapsed > TimeSpan.FromSeconds(5))
+                    {
+                        // If we still can't create the folder after 5 seconds, assume we will not be able to create it.
+                        // Return null to indicate this provider cannot provide source for the document.
+                        return null;
```

Comment:
Fine with this approach.
But I look at the caller here https://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/MetadataAsSource/MetadataAsSourceFileService.cs,108
Looks like it assumes the return value of this method can't be null. (line 177)
So maybe we need to handle the failed case more properly?

Reply:
Ah, I misread the null check in the caller.

Reply:
➡️ Rewrote this to continue as it did before, but to just skip writing the file to disk if the location is not accessible. Roslyn will treat the result as an empty document.